packages/flutter_tools/test/src/mocks.dart

```diff
@@ -29,6 +33,190 @@ class MockApplicationPackageStore extends ApplicationPackageStore {
   );
 }
 
+/// A strategy for creating Process objects from a list of commands.
+typedef Process ProcessFactory(List<String> command);
+
+/// A ProcessManager that starts Processes by delegating to a ProcessFactory.
+class MockProcessManager implements ProcessManager {
+  ProcessFactory processFactory = (List<String> commands) => new MockProcess();
+  List<String> commands;
+
+  @override
+  Future<Process> start(
+    List<dynamic> command, {
+    String workingDirectory,
+    Map<String, String> environment,
+    bool includeParentEnvironment: true,
+    bool runInShell: false,
+    ProcessStartMode mode: ProcessStartMode.NORMAL,
+  }) {
+    commands = command;
+    return new Future<Process>.value(processFactory(command));
+  }
+
+  @override
+  dynamic noSuchMethod(Invocation invocation) => null;
+}
+
+/// A process that exits successfully with no output and ignores all input.
+class MockProcess extends Mock implements Process {
+  MockProcess({
+    this.pid: 1,
+    int exitCode,
+    Stream<List<int>> stdin,
+    this.stdout: const Stream<List<int>>.empty(),
+    this.stderr: const Stream<List<int>>.empty(),
+  }) : exitCode = exitCode ?? new Future<int>.value(0),
```

Comment:
Is this legal? Assigning an int (exitCode) to Future<int>?

Reply:
Good catch -- Should take a `Future<int>`. Done.